Public/Start-ControlRemoteSupport.ps1

```diff
@@ -0,0 +1,71 @@
+function Start-ControlRemoteSupport {
+    <#
+    .SYNOPSIS
+        Will open a ConnectWise Control Remote Support session against a given machine.
+    .DESCRIPTION
+        Will open a ConnectWise Control Remote Support session against a given machine.
+
+    .PARAMETER ComputerName
+        The Automate computer name to connect to
+    .PARAMETER ComputerID
+        The Automate ComputerID to connect to
+    .PARAMETER ID
+        Taken from the Pipeline, IE Get-AutomateComputer -ComputerID 5 | Start-ControlRemoteSupport
+    .PARAMETER ComputerObjects
+        Used for Pipeline input from Get-AutomateComputer
+    .OUTPUTS
+        None (opens a Connect Control Remote Support session URL, via a URL to the default browser)
+    .NOTES
+        Version:        1.0
+        Author:         Jason Rush
+        Creation Date:  2019-10-15
+        Purpose/Change: Initial script development
+
+    .EXAMPLE
+        Start-ControlRemoteSupport -ComputerName TestComputer
+    .EXAMPLE
+        Start-ControlRemoteSupport -ComputerId 123
+    .EXAMPLE
+        Get-AutomateComputer -ComputerID 5 | Start-ControlRemoteSupport
+    #>
+    [CmdletBinding(DefaultParameterSetName = 'ID')]
+    param
+    (
+        [Parameter(ParameterSetName = 'Name', Mandatory = $true, Position = 0, ValueFromPipeline = $true, ValueFromPipelineByPropertyName=$False)]
+        [string[]]$ComputerName,
+
+        [Parameter(ParameterSetName = 'ID', Mandatory = $true, Position = 0, ValueFromPipeline = $true, ValueFromPipelineByPropertyName=$False)]
+        [int16[]]$ComputerID,
+
+        [Parameter(ParameterSetName = 'pipeline', ValueFromPipelineByPropertyName=$true, Mandatory = $True)]
+        [int16[]]$ID,
+
+        [Parameter(ParameterSetName = 'pipeline', ValueFromPipeline = $true, Mandatory = $True)]
+        $ComputerObjects
+        
+    )
+
+    Process {
+        #If not pipeline mode, build ComputerObjects
+        If ( ($PSCmdlet.ParameterSetName -eq 'ID') -or ($PSCmdlet.ParameterSetName -eq 'Name') ) {
+            $ComputerObjects = @()
+        }
+
+        If ($PSCmdlet.ParameterSetName -eq 'ID') {
+            ForEach ($ComputerIDSingle in $ComputerID) {
+                $ComputerObjects += (Get-AutomateComputer -ComputerID $ComputerIDSingle)
+            }
+        }
+
+        If ($PSCmdlet.ParameterSetName -eq 'Name') {
+            ForEach ($ComputerNameSingle in $ComputerName) {
+                $ComputerObjects += (Get-AutomateComputer -ComputerName $ComputerNameSingle)
+            }
+        }
+
+        ForEach ($Computer in $ComputerObjects) {
+            Start-Process "$((Get-AutomateComputer -ComputerID $Computer.ID | Get-AutomateControlInfo).LaunchURL)"
```

Comment:
Looks like this line could be replaced with:
```
            $ControlInfo = (Get-AutomateComputer -ComputerID $Computer.ID | Get-AutomateControlInfo)
            $ControlInfo.LaunchSession()
```
While there wouldn't be any performance improvement over what it's doing now, it would be more future-proof in case someone comes up with a better way to launch sessions, we only have to update in Get-AutomateControlInfo and not both places.

If Get-AutomateControlInfo were updated to actually get all the computer info, and not just a SessionID when run as `Get-AutomateControlInfo -ComputerID 1234`, we could make this line even cleaner, but it still wouldn't be any more efficient since the call to get the computer info would just be relocated to the other function.

Reply:
Speaking with @DarrenWhite99 the Get-AutomateComputer is redundant. Should be:
```
            $ControlInfo = Get-AutomateControlInfo -ComputerID $Computer.ID
            $ControlInfo.LaunchSession()
```